notification/rule/rule.go

```diff
@@ -245,12 +245,22 @@ func (b *Base) generateStateChanges(r notification.StatusRule) (ast.Statement, *
 	return flux.DefineVariable(name, pipe), flux.Identifier(name)
 }
 
+// increaseDur increases the duration of leading duration in a duration literal.
+// It is used so that we will have overlapping windows. If the unit of the literal
+// is `s`, we double the interval; otherwise we increase the value by 1. The reason
+// for this is to that we query the minimal amount of time that is likely to have data
+// in the time range.
 func increaseDur(d *ast.DurationLiteral) *ast.DurationLiteral {
 	dur := &ast.DurationLiteral{}
 	for i, v := range d.Values {
 		value := v
 		if i == 0 {
-			value.Magnitude++
+			switch v.Unit {
+			case "s":
```

Comment:
Minutes should hit the default case which is to increment by 1. The reasoning is that most reported metrics are reported on a second level basis and so doubling the frequency should work. For minutes, it'll hit the default case which is to increase the value by 1 since that should include the data.

The reason is to minimize the amount of data we include in the window.